_chapters/02_category.md

```diff
@@ -301,7 +303,7 @@ Note that from this property it follows that the singleton set has exactly one e
 
 The empty set is the set that has no elements, but how would we say this without referring to elements?
 
-We said that there exists a unique function that goes _from_ the empty set _to_ any other set. But the reverse is also true: the empty set is the only set such that there exists a function from it to any other set.
+We said that there exists a unique function that goes _from_ the empty set _to_ any other set. Or stated differently: the empty set is the only set such that there exists a function from it to any other set. In category theory we name this set the _initial object_ (as tracing all arrows backward to their source, we would eventually get here).
```

Comment:
"stated differently"  is not correct here, this is not one statement stated in two different ways, but two different statements:

1. singleton set -> terminal object 
2. terminal object -> singleton set

which together establishes the equivalence

terminal object <-> singleton set

(btw "terminal", not "initial")

Reply:
Thanks! I do understand this (i hope!).

But I am still pretty convinced after re-reading a few times that the two sentences are both referring to the empty set in the original text, and therefore confusing to say "the reverse is also true":

Original text:
> We said that there exists a unique function that goes _from_ the empty set _to_ any other set. But the reverse is also true: the empty set is the only set such that there exists a function from it to any other set.

(These both refer quite clearly to the empty set and therefore the initial object)

Could you maybe take another look?

Reply:
Yes, I don't think the original formulation is apt either, i was just thinking aloud. As usual, the "fix" is to elaborate further. 

(That is true of many other places btw, almost everywhere even. The main issue is that elaborating is so hard to do that I don't have the time and energy to fix all of them.)

Let's try:

In the previous chapter, we said that there exists a unique function that goes _from_ the empty set _to_ any other set. Another fact that is true is that the empty set is the only set for which this statements is true.

For this reason, we can reverse the above statement and use it as a definition: the empty set is the set such that there exists a function from it to any other set.

Task: why is the functions to the empty set unique?

Observant readers will notice the similarities between the *diagrams depicting the empty set and the singleton set* (yes the two concepts are, of course, dual of each other).

Some even more observant readers may also notice the similarities between this pair of definitions and the definitions of the product/coproduct * --- in both cases we find a property that makes a given concept useful and then define the concept by this property*.

(now I saw that I am using the terms initial terminal object without defining them -- better to just not use these terms instead of trying to define them too early).

Reply:
> In the previous chapter, we said that there exists a unique function that goes from the empty set to any other set.

There isn't any mention of the empty set anywhere in the previous chapter actually. We are introducing the empty set for the first time here (the previous chapter was focused on the singleton set / terminal object).

> (now I saw that I am using the terms initial terminal object without defining them -- better to just not use these terms instead of trying to define them too early).

Yep I noticed this and is why I added a single sentence naming each at the relevant paragraph (terminal object in the previous section and initial object here). Otherwise the first time we see these names is in the next diagram (which is excellent by the way).

Reply:
I think I will just revert this particular section and leave any changes to you - there are numerous approaches to make it clearer and I think our discussion here isn't useful. Will revert and push an update shortly.

Reply:
Done. Let me know if any other bits you would like to revert (just pop a comment at the line) - most are small typos.